controllers/configurationpolicy_utils_test.go

```diff
@@ -197,3 +198,113 @@ func TestEqualObjWithSortEmptyMap(t *testing.T) {
 	assert.True(t, equalObjWithSort(mergedObj, oldObj, true))
 	assert.False(t, equalObjWithSort(mergedObj, oldObj, false))
 }
+
+func TestGenerateDiff(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		existingObj  map[string]interface{}
+		updatedObj   map[string]interface{}
+		expectedDiff string
+		expectedErr  string
+	}{
+		"same object generates no diff": {
+			existingObj: map[string]interface{}{
+				"cities": map[string]interface{}{},
+			},
+			updatedObj: map[string]interface{}{
+				"cities": map[string]interface{}{},
+			},
+		},
+		"object with new child key": {
+			existingObj: map[string]interface{}{
+				"cities": map[string]interface{}{},
+			},
+			updatedObj: map[string]interface{}{
+				"cities": map[string]interface{}{
+					"raleigh": map[string]interface{}{},
+				},
+			},
+			expectedDiff: `
+@@ -1,2 +1,3 @@
+-cities: {}
++cities:
++  raleigh: {}`,
+		},
+		"object with new key": {
+			existingObj: map[string]interface{}{
+				"cities": map[string]interface{}{},
+			},
+			updatedObj: map[string]interface{}{
+				"cities": map[string]interface{}{},
+				"states": map[string]interface{}{},
+			},
+			expectedDiff: `
+@@ -1,2 +1,3 @@
+ cities: {}
++states: {}`,
+		},
+		"array with added item": {
+			existingObj: map[string]interface{}{
+				"cities": []string{
+					"Raleigh",
+				},
+			},
+			updatedObj: map[string]interface{}{
+				"cities": []string{
+					"Raleigh",
+					"Durham",
+				},
+			},
+			expectedDiff: `
+@@ -2,2 +2,3 @@
+ - Raleigh
++- Durham`,
+		},
+		"array with removed item": {
+			existingObj: map[string]interface{}{
+				"cities": []string{
+					"Raleigh",
+					"Durham",
+				},
+			},
+			updatedObj: map[string]interface{}{
+				"cities": []string{
+					"Raleigh",
+				},
+			},
+			expectedDiff: `
+@@ -2,3 +2,2 @@
+ - Raleigh
+-- Durham`,
+		},
+	}
+
+	for testName, test := range tests {
+		test := test
+
+		t.Run(testName, func(t *testing.T) {
+			t.Parallel()
+
+			existingObj := &unstructured.Unstructured{
+				Object: test.existingObj,
+			}
+			updatedObj := &unstructured.Unstructured{
+				Object: test.updatedObj,
+			}
+
+			diff, err := generateDiff(existingObj, updatedObj)
+			if err != nil {
+				assert.EqualError(t, err, test.expectedErr)
```

Comment:
Is there a way to test the error cases?

Reply:
I'll take a look. I don't think I can hit all of them, but I might be able to pass it a `nil` somewhere...

Reply:
I haven't worked out a way to hit the errors. I think the likelihood that the `generateDiff` function is passed a `nil` pointer is minimal/nonexistent (and would even require another check). Beyond that, it seems `yaml.Marshal()` will always (?) work on a `map[string]interface{}` Go object, and I think the `go-difflib` would likely only fail with system errors if the buffer fails, which also seems unlikely.

Unless anyone has other ideas on how to hit them, I'll remove this assertion and use `Fatal()` instead.